frontend/src/components/wallet/useWalletProvider.ts

```diff
@@ -172,12 +199,44 @@ export const useWalletProvider = () => {
     });
   };
 
+  const walletAddToken = async (
```

Comment:
Since this doesn't do anything with state, let's not include it in the provider. It can be a standalone utility.

Make a new file at `frontend/src/utils/metamask.ts` and throw it in there. This whole provider needs to get refactored.

Reply:
hmmmm good point